From 2df5c01047832a347b6f110ccfa0ceda581a3618 Mon Sep 17 00:00:00 2001
From: tanxiaofei <tanxiaofei@huawei.com>
Date: Tue, 19 Dec 2017 09:51:45 +0800
Subject: [PATCH 1/2] add support of hisi sas v2 hw

Signed-off-by: Xiaofei Tan <tanxiaofei@huawei.com>
---
 src/cntrl.c | 15 +++++++++++++++
 src/sysfs.c |  3 ++-
 2 files changed, 17 insertions(+), 1 deletion(-)

diff --git a/src/cntrl.c b/src/cntrl.c
index 29195d1..c35c067 100644
--- a/src/cntrl.c
+++ b/src/cntrl.c
@@ -84,6 +84,19 @@ static int _is_dellssd_cntrl(const char *path)
 	        (svdr == 0x1028 && cls == 0x10802));  /* nvmhci ssd */
 }
 
+static int _is_hisi_cntrl(const char *path)
+{
+	uint32_t str = 0;
+	char *text =NULL;
+
+	text = get_text(path, "uevent");
+	if (text) {
+		str = strstr(text, "hisi_sas");
+		free(text);
+	}
+	return str;
+}
+
 /**
  */
 static int _is_smp_cntrl(const char *path)
@@ -200,6 +213,8 @@ static enum cntrl_type _get_type(const char *path)
 		type = CNTRL_TYPE_DELLSSD;
 	} else if (_is_vmd_cntrl(path)) {
 		type = CNTRL_TYPE_VMD;
+	} else if (_is_hisi_cntrl(path) && _is_smp_cntrl(path)) {
+		type = CNTRL_TYPE_SCSI;
 	} else if (_is_storage_controller(path)) {
 		if (_is_ahci_cntrl(path))
 			type = CNTRL_TYPE_AHCI;
diff --git a/src/sysfs.c b/src/sysfs.c
index c6e8205..8d34840 100644
--- a/src/sysfs.c
+++ b/src/sysfs.c
@@ -47,6 +47,7 @@
  */
 #define SYSFS_CLASS_BLOCK       "/sys/block"
 #define SYSFS_CLASS_ENCLOSURE   "/sys/class/enclosure"
+#define SYSFS_PLATFORM_DEVICES       "/sys/devices/platform"
 #define SYSFS_PCI_DEVICES       "/sys/bus/pci/devices"
 #define SYSFS_PCI_SLOTS         "/sys/bus/pci/slots"
 
@@ -439,7 +440,7 @@ static status_t _scan_raid(void)
  */
 static status_t _scan_cntrl(void)
 {
-	void *dir = scan_dir(SYSFS_PCI_DEVICES);
+	void *dir = scan_dir(SYSFS_PLATFORM_DEVICES);
 	if (dir) {
 		list_for_each(dir, _check_cntrl);
 		list_fini(dir);
-- 
1.9.1

